src/EnergyPlus/ZoneTempPredictorCorrector.cc

```diff
@@ -3608,7 +3608,10 @@ namespace ZoneTempPredictorCorrector {
                 if (TempCtrlFound && TempControlledZone(RelativeZoneNum).DeltaTCutSet > 0.0) {
                     TempZoneThermostatSetPoint(ZoneNum) = TempControlledZone(RelativeZoneNum).ZoneThermostatSetPointLo;
                     ZoneThermostatSetPointLo(ZoneNum) = TempControlledZone(RelativeZoneNum).ZoneThermostatSetPointLo;
-                    if (ZoneAirSolutionAlgo == Use3rdOrder) Tprev = ZTM1(ZoneNum);
+                    if (ZoneAirSolutionAlgo == Use3rdOrder) {
+                        Tprev = MAT(ZoneNum);
+                        if (ShortenTimeStepSys) Tprev = XMPT(ZoneNum);
+                    }
```

Comment:
Instead of testing for shortened time step could ZTM1 be updated correctly so this test isn't needed here?

Reply:
@rraustad You raised a good question. Here are several reasons I am reluctant to make changes of ZTM1 correctly in the original place.

1. ZTM1 is assigned by XMAT in line 3190 in ZoneTempPredictorCorrector
                ZTM1(ZoneNum) = XMAT(ZoneNum);
XMAT is assigned by ZTAV in line 4911 in ZoneTempPredictorCorrector
            XMAT(ZoneNum) = ZTAV(ZoneNum); // using average for whole zone time step.

2 When DownInterpolate4HistoryValues is called, the zone air temperature at previous time steps used in 3order equations are interpolated with lengths of previous time steps.

Therefore, ZTM1 is not exactly a value at previous time step.

The best way to fix is to remove downshifting, so that we don't have any issues for history terms. I proposed this idea before. 

My fix is to allow zone air temperature at previous time step is used correctly for this new feature only. If I change it in the originally place, it will cause many differences, so that it will not be a small fix.